### ISSUE TITLE
Smarter package management

### ISSUE THREAD

**Issue by Jint-lzxy** [ayamir/nvimdots#1293]
### Feature description

As briefly discussed in #1252, there are several cases where Mason is simply dead and can't be used at all. My specific situation is that most of the pre-built binaries can't run on FreeBSD ~~(of course)~~, and Linux-emulation mode doesn't really help either. I need to constantly symlink the "correct" executables to Mason's `bin` directory to make things work, but that's laborious, and I really don't wanna do that anymore. imo we need a way to either bypass Mason completely or make it optional for OSs like NixOS and the BSDs.

**Comment by vollowx** [comment]
The LSP servers, linters and formatters can actually be installed in system-wide on most systems, which means they are executable everywhere and maybe we just don't need mason at all?

**Comment by Jint-lzxy** [comment]
> The LSP servers, linters and formatters can actually be installed in system-wide on most systems, which means they are executable everywhere and maybe we just don't need mason at all?

Exactly! Initially, I was thinking of completely removing Mason, but without it, we'd lose the ability to automatically install those dependencies on a freshly installed system, which is kinda like the two sides of the same coin lol

**Comment by Jint-lzxy** [comment]
> Maybe ask user whether to install stuffs manually or by mason, store the choice and enable mason if that is `true`?

Yeah that's what I meant by "optional," but ideally, I'd like to find a way to automatically _bypass_ Mason without any user interaction. In my case, I'd love Mason to be available on my Mac, but to be automatically disabled on my FreeBSD workstation. Especially in [jails](https://docs.freebsd.org/en/books/handbook/jails/) when I'm doing certain experimentation, I certainly don't wanna specify that option manually lmao

**Comment by CharlesChiuGit** [comment]
> In my case, I'd love Mason to be available on my Mac, but to be automatically disabled on my FreeBSD workstation. 

Maybe we can add an option as "platform" in our custom `lazy.nvim` loader?
If `platform = {"XXX OS" = false}` is not set, then it will default to install the plugin on whatever OS it's on.

**Comment by Jint-lzxy** [comment]
> Maybe we can add an option as "platform" in our custom `lazy.nvim` loader and utilized the `enabled` flag in `lazy.nvim`? If `platform = {"XXX OS" = false}` is not set, then it will default to install the plugin on whatever OS it's on. I personally would like to disable `sniprun` on windows since it could not be installed on windows lol

lol that's a great idea! Tho we'll need a way to automatically register those sources with lspconfig/none-ls/dap then.

**Comment by Jint-lzxy** [comment]
After some consideration, I've decided to create a plugin that automates the control of those "bridging plugins" (like `mason-null-ls.nvim`). I'll share the link to the Reddit post here once it's completed lol

**Comment by hiSandog** [comment]
One split that may keep this manageable is to separate "tool discovery" from "tool installation". Mason could remain one installer backend, while the runtime resolver first checks explicit user paths/system executables and only falls back to Mason-managed binaries when enabled. That would make FreeBSD/NixOS/BSD setups first-class without removing the fresh-install convenience for users who want Mason to bootstrap everything.

**Comment by charliie-dev** [comment]
~~@hiSandog i just make a PR to support this, u can check it if u feel like to~~

**Comment by charliie-dev** [comment]
https://github.com/charliie-dev/muster.nvim
i instead making a plugin for this feature. still very much in alpha.
i'll test this plugin on my own fork first.

**Comment by charliie-dev** [comment]
some screenshot from muster.nvim lol

<img width="695" height="406" alt="Image" src="https://github.com/user-attachments/assets/f3b53b88-8f69-4411-b389-bdb8f23b50b4" />

<img width="1398" height="907" alt="Image" src="https://github.com/user-attachments/assets/91ef7698-3a37-496b-b181-620119f070d7" />
